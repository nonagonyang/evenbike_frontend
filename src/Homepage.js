function Homepage() {
  return (
    <>
      <h1>HomePage EvenBike</h1>
    </>
  );
}

export default Homepage;
