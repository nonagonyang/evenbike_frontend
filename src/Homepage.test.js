import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import Homepage from "./Homepage";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";

it("renders without crashing", function () {
  render(
    <BrowserRouter>
      <Homepage />
    </BrowserRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <BrowserRouter>
      <Homepage />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("works when you click on the Plan a trip button", async function () {
  const { getByText, getByLabelText } = render(
    <BrowserRouter>
      <Homepage />
    </BrowserRouter>
  );

  expect(getByText("Plan A Trip")).toBeInTheDocument();
  // const tripButton = getByText("Plan A Trip");

  // // click on the button
  // fireEvent.click(tripButton, new MouseEvent("click", { bubbles: true }));
  // // await waitFor(() => getByLabelText("Username"), {
  // //   timeout: 2000, // wait 2s
  // // });
  // // const username = getByText("Username");
  // // // expect the sign in page to show
  // // expect(username).toBeInTheDocument();
  // expect(getByText("Sign In")).not.toBeInTheDocument();
});
