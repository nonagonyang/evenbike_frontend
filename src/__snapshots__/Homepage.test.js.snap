// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="landing-page"
  >
    <div
      class="container"
    >
      <div
        class="header-area"
      >
        <div
          class="logo"
        >
          <b>
            EvenBike
          </b>
        </div>
      </div>
      <div
        class="info"
      >
        <h1>
          Greener Public Bike
        </h1>
        <p>
          Most public bikes have to be redistributed by trucks in the evening. Join us to help redistribute public bikes without using trucks.
        </p>
        <button>
          Plan A Trip
        </button>
      </div>
      <div
        class="image"
      >
        <img
          alt=""
          src="bike.jpg"
        />
      </div>
      <div
        class="clearfix"
      />
    </div>
  </div>
</DocumentFragment>
`;
